commandapi-platforms/commandapi-bukkit/commandapi-bukkit-vh/src/main/java/dev/jorel/commandapi/CommandAPIVersionHandler.java

```diff
@@ -68,8 +72,23 @@ public interface CommandAPIVersionHandler {
 				case "1.20.3", "1.20.4" -> new NMS_1_20_R3();
 				case "1.20.5", "1.20.6" -> new NMS_1_20_R4();
 				case "1.21", "1.21.1" -> new NMS_1_21_R1();
-				default -> throw new UnsupportedVersionException(version);
+				default -> null;
 			};
+			if (platform != null) {
+				return new LoadContext(platform, () -> {});
+			}
+			if (CommandAPI.getConfiguration().shouldBeLenientForMinorVersions()) {
+				int currentMajorVersion = Integer.parseInt(version.split("\\.")[1]);
+				if (latestMajorVersion == currentMajorVersion) {
+					return new LoadContext(new NMS_1_21_R1(), () -> {
+						CommandAPI.logWarning("Loading the CommandAPI with a potentially incompatible NMS implementation.");
+						CommandAPI.logWarning("While you may find success with this, further updates might be necessary to fully support the version you are using.");
+					});
+				} else {
+					throw new UnsupportedVersionException(version);
+				}
```

Comment:
Is this else block necessary? If it was removed, execution should continue to the `throw` statement currently on line 91, resulting in the same outcome.

```suggestion
				}
```